swan-lake/learn/testing-ballerina-code/writing-tests.md

```diff
@@ -334,28 +373,105 @@ function beforeFunc() {
     io:println("I'm the before function!");
 }
 
-// Test function.
+// First test function.
 @test:Config {}
 function testFunction1() {
     io:println("I'm in test function 1!");
     test:assertTrue(true, msg = "Failed!");
 }
 
-// Test function.
+// Second test function.
 @test:Config {}
 function testFunction2() {
     io:println("I'm in test function 2!");
     test:assertTrue(true, msg = "Failed!");
 }
 
-// Test function.
+// Third test function.
+@test:Config {}
+function testFunction3() {
+    io:println("I'm in test function 3!");
+    test:assertTrue(true, msg = "Failed!");
+}
+```
+
+### @test:AfterEach
+
+The function specified after this annotation will be run after each test within the test suite is run. This can be
+ used for repeatedly cleaning up test-level aspects before every test function.
+
+***Example:***
+
+```ballerina
+import ballerina/io;
+import ballerina/test;
+
+// After each function, which is executed before each test function
+@test:AfterEach
+function beforeFunc() {
+    io:println("I'm the before function!");
+}
+
+// First test function
+@test:Config {}
+function testFunction1() {
+    io:println("I'm in test function 1!");
+    test:assertTrue(true, msg = "Failed!");
+}
+
+// Second test function
+@test:Config {}
+function testFunction2() {
+    io:println("I'm in test function 2!");
+    test:assertTrue(true, msg = "Failed!");
+}
+
+// Third test function
 @test:Config {}
 function testFunction3() {
     io:println("I'm in test function 3!");
     test:assertTrue(true, msg = "Failed!");
 }
 ```
 
+## @test:AfterGroups {}
+
+***Example:***
+
+For each group specified in this annotation, the function that follows the annotation will be executed once after
+ all the tests belonging to the group is executed.
+
+```ballerina
+import ballerina/io;
+import ballerina/test;
+
+// A test function that belongs to the group `g1`
+@test:Config { groups:["g1"] }
+function testFunction1() {
+    io:println("I belong to group g1!");
+    test:assertTrue(true, msg = "Failed");
+}
+
+// A test function that belongs to the group `g2`
+@test:Config { groups:["g2"] }}
+function testFunction2() {
+    io:println("I belong to group g2 ");
+    test:assertTrue(true, msg = "Failed");
+}
+
+// The `AfterGroups` function is executed before running all the test functions in this module. 
+@test:BeforeGroups { value:["g1"] }
+function beforeFunc() {
+    io:println("I'm the before groups function!");
+}
+
+// Another `AfterGroups` function is executed before running all the test functions in this module. 
+@test:BeforeGroups { value:["g1", "g2"] }
+function beforeFunc() {
+    io:println("I'm another before groups function!");
+}
+```
+
 ### @test:AfterSuite {}
 
 The function specified after the annotation will be run once after all of the tests in the test suite is run. This can be used for cleaning up the test suite level aspects. The test suite covers tests related to a module.
```

Comment:
```suggestion
The function specified after the annotation will be run once after all of the tests in the test suite are run. This can be used for cleaning up the test suite level aspects. The test suite covers tests related to a module.
```